File: Algorithms/11. Container With Most Water/swift/solution.swift
class Solution {
  func maxArea(_ height: [Int]) -> Int {
    var left = 0
    var right = height.count - 1
    var area = 0

    while left < right {
      area = max(area, min(height[left], height[right]) * (right - left))

      if height[right] > height[left] {
        left += 1
      } else {
        right -= 1
      }
    }

    return area
  }
}

File: Algorithms/11. Container With Most Water/python/solution.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        area = 0
        left = 0
        right = len(height) - 1

        while left < right:
            area = max(area, min(
                height[left], height[right]) * (right - left))

            if height[right] > height[left]:
                left += 1
            else:
                right -= 1

        return area
